ingest/src/withConfig.ts

```diff
@@ -0,0 +1,88 @@
+import yargs from "yargs";
+import Path from "path";
+import { Config, Constructor } from "./Config";
+
+export type LoadConfigArgs = {
+  config?: string;
+};
+
+export const loadConfig = async ({
+  config: configPathIn,
+}: LoadConfigArgs): Promise<Config> => {
+  const path = Path.resolve(
+    configPathIn === undefined ? "ingest.config.cjs" : configPathIn
+  );
+
+  const configArray = (await import(path)).default as Config[];
+
+  // Validate config
+  if (!Array.isArray(configArray)) {
+    throw new Error(
+      `Invalid config at ${path}: expected default exported array of Config objects`
+    );
+  }
+  // TODO: further validation
```

Comment:
@nlarew did some nice type validation stuff on the server w zod. wonder if we could do same here.